urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/experiments/ScalabilityScenarioGenerator.java

```diff
@@ -0,0 +1,229 @@
+import br.edu.ufcg.lsd.oursim.entities.grid.Grid;
+import br.edu.ufcg.lsd.oursim.entities.grid.Peer;
+import br.edu.ufcg.lsd.oursim.entities.grid.Worker;
+import br.edu.ufcg.lsd.oursim.entities.job.ExecutionState;
+import br.edu.ufcg.lsd.oursim.entities.job.Job;
+import br.edu.ufcg.lsd.oursim.events.Event;
+import br.edu.ufcg.lsd.oursim.events.EventSpec;
+import br.edu.ufcg.lsd.oursim.events.broker.BrokerEvents;
+import br.edu.ufcg.lsd.oursim.events.ds.DiscoveryServiceEvents;
+import br.edu.ufcg.lsd.oursim.events.global.HaltCondition;
+import br.edu.ufcg.lsd.oursim.events.peer.PeerEvents;
+import br.edu.ufcg.lsd.oursim.events.worker.WorkerEvents;
+import br.edu.ufcg.lsd.oursim.fd.FailureDetectorOptInjector;
+import br.edu.ufcg.lsd.oursim.network.DefaultWANNetwork;
+import br.edu.ufcg.lsd.oursim.queue.ListEventProxy;
+import br.edu.ufcg.lsd.oursim.trace.DefaultTraceCollector;
+import br.edu.ufcg.lsd.oursim.util.Configuration;
+
+public class ScalabilityScenarioGenerator {
+
+	private static final String BROKER_ID = "brokerA";
+	private static final int REPLICATION = 1;
+	static int TASKS_DURATION = 10000; 
+	static double LOCALHOST_DELAY = 5.;
+	
+	static int WORKERS_PER_SITE = 50;
+	static int CONTENTION_LEVEL = 1;
+	
+	static int[] SITES = new int[]{1, 5, 10, 50, 100, 500, 1000, 5000, 10000};
+	static int JOBS_PER_BROKER = 1;
+	
+	static String SCENARIO = "scalability";
+	
+	static int[][] SEEDS = new int[][]{
+		new int[]{214338003, 944166635, 570782134, 526559949, 919594043, 214338003}, 
+		new int[]{413630306, 617652072, 984949403, 2098010699, 1519794858, 413630306}, 
+		new int[]{70395995, 2035636432, 1898631199, 1140526672, 787341018, 70395995}, 
+		new int[]{1146986511, 1886216422, 282759243, 1356582152, 538696788, 1146986511}, 
+		new int[]{1303202085, 1074684734, 2004358020, 30184504, 1102953255, 1303202085} 
+	};
+	
+	public static void main(String[] args) throws Exception {
+	
+		Properties properties = Configuration.createConfiguration(new Properties());
+		properties.put(Configuration.PROP_BROKER_MAX_REPLICAS, "1");
+		properties.put(Configuration.PROP_BROKER_MAX_SIMULTANEOUS_REPLICAS, "1");
+		properties.put(Configuration.PROP_USE_FAILURE_DETECTOR, Boolean.TRUE.toString());
+		properties.put(Configuration.PROP_BROKER_SCHEDULER_INTERVAL, "10000");
+		properties.put(Configuration.PROP_FAILURE_DETECTOR_NAME, "fixed");
+		properties.put(Configuration.PROP_USE_SPEED_HACK, Boolean.TRUE.toString());
+		
+		for (int siteIdx = 0; siteIdx < SITES.length; siteIdx++) {
+			run(properties, SITES[siteIdx]);
+		}
+	}
+
+	private static void run(Properties properties, int sitesNo)
+			throws JSONException, IOException {
+		for (int i = 0; i < REPLICATION; i++) {
+			DefaultWANNetwork.setSeed(SEEDS[i]);
+			run(properties, WORKERS_PER_SITE, sitesNo, sitesNo * WORKERS_PER_SITE * CONTENTION_LEVEL, i);
+		}
+	}
+	
+	private static void run(Properties properties, int workersNo, int sitesNo, int tasksNo, int replica)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 51 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 146-175 into a separate method.